### PR TITLE
PAY-3831: helm chart upgrade and spring boot upgrade changes

### PR DESCRIPTION
**Pull request by prashant-khunte**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,14 +1,14 @@
 plugins {
     id 'application'
     id "org.sonarqube" version "3.0"
-    id 'org.springframework.boot' version '2.2.6.RELEASE'
+    id 'org.springframework.boot' version '2.3.0.RELEASE'
```
Comment (amitchidrewar1301) [inline build.gradle:4]: You need to upgrade to the latest 2.3.x version, 2.3.6 seems to be the latest one, so please update the version.

File: build.gradle [hunk 2]
```diff
@@ -118,6 +118,9 @@ subprojects {
         compileOnly group: 'org.projectlombok', name: 'lombok', version: versions.lombok
         annotationProcessor group: 'org.projectlombok', name: 'lombok', version: versions.lombok
         testAnnotationProcessor group: 'org.projectlombok', name: 'lombok', version: versions.lombok
+
+        implementation 'org.springframework.boot:spring-boot-starter-validation'
```
Comment (amitchidrewar1301) [inline build.gradle:122]: Please try by removing this dependency

Comment (Jennings-HMCTS) [reply build.gradle:133]: Certain annotations like NotNull depends on this package. So this dependency is required

File: api/build.gradle [hunk 3]
```diff
@@ -74,11 +85,11 @@ dependencies {
   compile group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: javaLoggingVersion
   compile group: 'uk.gov.hmcts.reform', name: 'logging-spring', version: javaLoggingVersion
   compile group: 'uk.gov.hmcts.reform.auth', name: 'auth-checker-lib', version: '2.1.3'
-  compile group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: versions.reformS2sClient
+  compile (group: 'uk.gov.hmcts.reform', name: 'service-auth-provider-client', version: versions.reformS2sClient)
```
Comment (amitchidrewar1301) [inline api/build.gradle:88]: Please revert to previous changes, remove brackets, extra spaces from api/build.gradle

File: dependency-check-suppressions.xml [hunk 4]
```diff
@@ -98,10 +103,12 @@
         <notes>HTTP Client 4.5.11 suppression, root cause and fix needs to be carried out separately</notes>
         <cve>CVE-2020-13956</cve>
     </suppress>
+
```
Comment (amitchidrewar1301) [inline dependency-check-suppressions.xml:106]: Please remove the new line changes

File: api/build.gradle (hunk 3 above)
Comment (Jennings-HMCTS) [reply api/build.gradle]: changes made

File: dependency-check-suppressions.xml (hunk 4 above)
Comment (Jennings-HMCTS) [reply dependency-check-suppressions.xml:106]: made required changes

File: charts/payment-api/Chart.yaml [hunk 5]
```diff
@@ -1,9 +1,13 @@
...
 name: payment-api
 home: https://github.com/hmcts/ccpay-payment-app
-version: 2.0.35
+version: 2.0.36
 maintainers:
   - name: HMCTS Fees & Payments Dev Team
     email: ccpay@hmcts.net
+dependencies:
+  - name: java
+    version: ~3.0.0-beta
```
Comment (timja) [inline charts/payment-api/Chart.yaml:12]: don't use beta versions

File: charts/payment-api/values.yaml [hunk 6]
```diff
@@ -1,6 +1,7 @@
 java:
   image: hmctspublic.azurecr.io/payment/api:latest
   applicationPort: 8080
+  readinessPath: '/health/readiness'
```
Comment (timja) [inline charts/payment-api/values.yaml:4]: this is the default on latest spring and the helm chart
```suggestion
```

File: dependency-check-suppressions.xml [hunk 7]
```diff
@@ -89,6 +89,11 @@
         <cve>CVE-2020-15250</cve>
     </suppress>
 
+    <suppress>
+        <notes>This suppress specific cve for Jetty Library which is a transitive dependency</notes>
```
Comment (timja) [inline dependency-check-suppressions.xml:93]: transitive doesn't matter you still have it, it should be patched if available or at a minimum needs an expiry date

**Review by timja — CHANGES_REQUESTED** [review]
CVE and chart-java issue

File: dependency-check-suppressions.xml (hunk 7 above)
Comment (Jennings-HMCTS) [reply dependency-check-suppressions.xml:93]: @timja This transitive dependency also has CVE issues and hence this supression is added

**Review by timja — APPROVED** [review]
You should be able to set the version resolution in an allprojects block I think and only do it once, but no blockers from me

**Review by aravindan89 — APPROVED** [review]

**Review by YasharRahvar — APPROVED** [review]

**Review by YasharRahvar — APPROVED** [review]

**Pull request merged**